interlok-core/src/main/java/com/adaptris/core/management/config/JavaxValidationChecker.java

```diff
@@ -0,0 +1,46 @@
+package com.adaptris.core.management.config;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import com.adaptris.core.Adapter;
+import com.adaptris.core.CoreException;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+public class JavaxValidationChecker extends AdapterConfigurationChecker {
+
+  private static final String FRIENDLY_NAME = "javax.validation Checks";
+  private static final ValidatorFactory JAVAX_VALIDATOR_FACTORY =
+      Validation.buildDefaultValidatorFactory();
+
+
+  @Override
+  protected void validate(Adapter adapter, ConfigurationCheckReport report) {
+    try {
+      Validator validator = JAVAX_VALIDATOR_FACTORY.getValidator();
+      // There are no warnings; everything is a hard failure.
+      report.setFailureExceptions((violationsToException(validator.validate(adapter))));
+    } catch (Exception ex) {
+      report.getFailureExceptions().add(ex);
+    }
+  }
+
+  @Override
+  public String getFriendlyName() {
+    return FRIENDLY_NAME;
+  }
+
+  private List<Exception> violationsToException(Set<ConstraintViolation<Adapter>> violations) {
+    List<Exception> result = new ArrayList<>();
```

Comment:
Not a one liner?

```
  private List<Exception> violationsToException(Set<ConstraintViolation<Adapter>> violations) {
    return violations.stream().map(this::violationsCoreException).collect(Collectors.toList());
  }

  private CoreException violationsCoreException(ConstraintViolation<?> violation) {
    return new CoreException(
        String.format("Interlok Validation Error: [%1$s]=[%2$s]", violation.getPropertyPath(), violation.getMessage()));
  }
```

Reply:
You're right, but I had different issues with mocking ;) -> I shall change that to use a lambda (however, I haven't really loved the :: syntax so it might really be a one line (!).

Reply:
https://github.com/adaptris/interlok/pull/497/commits/f92fac6d171187cad1163e3c25952e08e28f8fa9